packages/cli/src/lib/Prompt.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {Result} from '@bubblewrap/core';
+import * as inquirer from 'inquirer';
+
+export type ValidateFunction<T> = (input: string) => Result<T, Error>;
+
+export interface Prompt {
+  /**
+   * Prompts for free text input
```

Comment:
nit: Maybe I haven't noticed this before, but shouldn't our comments all be complete sentences (with a full stop)?

Reply:
No, that's my bad - I keep forgetting about the full stops (I'll see if there's a lint rule for that). Fixed.